### DISCUSSION TITLE
completion-replace and terraform-ls

### DISCUSSION THREAD

**Discussion by anikhad** [helix-editor/helix#6318] [General]
I am trying to get autocomplete to work for terraform-ls in a way that the entire word is replaced after tabbing to the selection. I am on version helix 22.12 (96ff64a8) installed using brew. I get the following behaviour when I try to set completion-replace = true in my config.toml. Does this parameter work with a later version of helix?

code $ hx
Bad config: unknown field `completion-replace`, expected one of `scrolloff`, `scroll-lines`, `mouse`, `shell`, `line-number`, `cursorline`, `cursorcolumn`, `gutters`, `middle-click-paste`, `auto-pairs`, `auto-completion`, `auto-format`, `auto-save`, `idle-timeout`, `completion-trigger-len`, `auto-info`, `file-picker`, `statusline`, `cursor-shape`, `true-color`, `search`, `lsp`, `terminal`, `rulers`, `whitespace`, `bufferline`, `indent-guides`, `color-modes` for key `editor` at line 22 column 1
Press <ENTER> to continue with default config

My config looks like:
theme = "nord"

[editor]
line-number = "relative"
auto-completion = true
completion-replace = true
gutters = ["diagnostics", "diff", "line-numbers"]
auto-format = true
idle-timeout = 100
completion-trigger-len = 1
bufferline = "multiple"

[editor.cursor-shape]
insert = "bar"
normal = "block"
select = "block"

[editor.statusline]
left = ["spinner","mode", "file-name"]
right = ["position","file-type","file-encoding"]

A second issue that may not be a helix issue but a terraform-ls issue is that autocomplete does not work as expected for certain completions. E.g.
terraform {
  required_providers {
    azurerm = {
      **ssource = "hashicorp/azurerm"
      vversion = "3.47.0"**
    }
  }
}

It works as expected for top level keywords but for **source** and **version** it appends to the trigger alphabet (s and v since I've set it to len 1). Is this a helix issue or a terraform-ls issue? Are there configs to set in languages.toml to influence this behaviour?

Thank you for having a look at this question.

**Comment by kirawi** [comment]
That option was recently added to master, so it's not in a release yet.
